.ci/release/Jenkinsfile

```diff
@@ -0,0 +1,235 @@
+                def ret = githubReleaseCreate(draft: true, body: "[Release Notes for $bareTag]($finalUrl)")
+                env.RELEASE_ID = ret['id']
+              }
+            }
+
+          }
+        }
+        // 10. Update cloudfoundry/index.yml
+        // Needs to append a line such as the following:
+        // 1.13.0: https://repo1.maven.org/maven2/co/elastic/apm/elastic-apm-agent/1.13.0/elastic-apm-agent-1.13.0.jar
+        stage('Update Cloudfoundry') {
+          steps {
+            dir("${basedir}") {
+              script {
+                sh("git checkout master")
+                def bareTag = sh(script: 'git tag|tail -1|sed s/v//', returnStdout: true)
+                def baseUrl = "https://repo1.maven.org/maven2/co/elastic/apm/elastic-apm-agent"
+                def lineToAppend = sh(script: "echo '$baseUrl: $baseUrl/$bareTag/elastic-apm-agent-$bareTag\.jar' >> cloudfoundry/index.yml ")
+                gitPush()
+              }
+            }
+          }
+        }
+        // 11. Wait for released package to be available in maven central
+        stage('Wait for artifact to be available in Maven Central') {
+          steps {
+            dir("${BASE_DIR}"){
+              script {
+                // We are looking for something like this:
+                // https://oss.sonatype.org/service/local/repositories/releases/content/co/elastic/apm/apm-agent-java/1.1.12/apm-agent-java-1.1.12.pom
+                def baseUrl = "https://oss.sonatype.org/service/local/repositories/releases/content/co/elastic/apm/apm-agent-java"
+                def bareTag = sh(script: 'git tag|tail -1|sed s/v//', returnStdout: true)
+                def fullUrl = sh(script: "$baseUrl/$bareTag/apm-agent-java-$bareTag\.pom")
+                waitUntil(initialRecurrencePeriod: 15000) {
+                  script {
+                    def ret = sh(script: "curl -fs $fullUrl >/dev/null 2>&1", returnStatus: true)
+                    echo "Waiting for the artifacts to be published on Sonatype"
+                    return ret == 0
+                  }
+                }
+              }
+            }
+
+          }
+        }
+        // 12. Publish release on Github. This will notify users watching repository.
+        stage('Publish release on GitHub') {
+          steps {
+            githubReleasePublish(id: ${env.RELEASE_ID})
+          }
+        }
+        // 13. Publish Docker images
+        stage('Docker push') {
+          when {
+            beforeAgent true
+            expression { return params.push_docker }
+          }
+          steps {
+            sh(label: "Build Docker image", script: "scripts/jenkins/build_docker.sh")
+            // Get Docker registry credentials
+            dockerLogin(secret: "${ELASTIC_DOCKER_SECRET}", registry: 'docker.elastic.co')
+            sh(label: "Push Docker image", script: "scripts/jenkins/push_docker.sh")
+          }
+        }
+        // 14. Opbeans
+        stage('Opbeans') {
```

Comment:
This should be tag-based, this pipeline is master based IIRC, therefore the tag conditional can be left as it was previously, what do you think?

Reply:
@v1v Do you mean this this conditional specifically?

```
      when {
        tag pattern: 'v\\d+\\.\\d+\\.\\d+', comparator: 'REGEXP'
      }
```

Reply:
Let me clarify, I meant, this particular stage could be kept in the other CI Jenkinsfile. IIUC, this particular pipeline will be triggered for the master branch and as a consequence, it will create a tag/branch with the version.

Therefore, the other CI Jenkinsfile could be kept as it was, it was configured to listen for branches/tags/PRs and the `when condition` in place will care of running that particular stage. 

I think it might help to simplify this pipeline. what do you think?

Reply:
Gotcha! I agree and have restored the original logic. Thanks!